### ISSUE TITLE
False positive: Style/RedundantConstantBase

### ISSUE THREAD

**Issue by jeffwidman** [rubocop/rubocop#11401]
## Expected behavior

Expected Rubocop to not complain about `Style/RedundantConstantBase` for methods:
```
let(:source) { ::Dependabot::Source.new(provider: provider, repo: repo) }
```

## Actual behavior

Rubocop flagged it as an issue.

## Steps to reproduce the problem

This is visible in CI on the open source Dependabot library, for the PR that bumps our Rubocop version up to `1.42.0`.

Some CI failures:
* https://github.com/dependabot/dependabot-core/actions/runs/3841072315/jobs/6540888749#step:5:12 (run rubocop within the `/updater` folder)
* https://github.com/dependabot/dependabot-core/actions/runs/3841073744/jobs/6540895107#step:6:184 (run rubocop within the `/bundler` folder)

So should be easily reproducible by git cloning that repo and running rubocop within the `/updater` or the `bundler` folder.

If needed, can also load this in our docker dev environment here: https://github.com/dependabot/dependabot-core#running-with-docker
But I suspect it will repro locally w/o needing docker etc.

## RuboCop version

`1.42.0`

**Comment by jeffwidman** [comment]
Similar false positive failures are here: https://github.com/dependabot/dependabot-core/actions/runs/3841073744/jobs/6540895107#step:6:184

**Comment by koic** [comment]
This looks like expected new warns rather than false positives. `Style/RedundantConstantBase` points to `::` which can be omitted. If you don't like that style you can disable `Style/RedundantConstantBase`.

**Comment by jeffwidman** [comment]
Well, it's either false positives, or `rubocop -a` has a bug... to repro:
1. `main` is passing CI: https://github.com/dependabot/dependabot-core/commit/42892139bd49e47a6cdabd9c91d2213ee38b7a62
2. The PR https://github.com/dependabot/dependabot-core/pull/6369 bumps Rubocop in https://github.com/dependabot/dependabot-core/pull/6369/commits/928b3d820e7c46d0138cdbf3493c4dfa621081a5 , which then causes CI to fail due to the `Style/RedundantConstantBase` linter error: https://github.com/dependabot/dependabot-core/actions/runs/3859906407/jobs/6579855642
3. So I ran `rubocop -a` on that branch in https://github.com/dependabot/dependabot-core/pull/6369/commits/ab85c1a657d017e944f86b031c036209a47a36fa, and it fixes the warnings, but now a unit test is failing (which previously passed before `rubocop -a` ran): https://github.com/dependabot/dependabot-core/actions/runs/3860198468/jobs/6580370809

**Comment by jeffwidman** [comment]
Here's something else interesting... the way this code is organized, we've got a library for Dependabot's core code, and then alongside it an app called "updater" that GitHub uses to run Dependabot.

The failing unit test I mentioned ☝️ is after I ran `rubocop -a` against the library, but the unit test failure is in the updater code.

When I run `rubocop -a` against the updater code, it fixes one instance of `::` in the updater code: https://github.com/dependabot/dependabot-core/pull/6389/commits/02858b8d8937804e851f68b5bfe1658db5115e7e

And then the unit test starts passing.

🤔 so that implies that we can make this all work if we update both the library and the apps calling it... but I don't think running `rubocop -a` against a library should introduce a breaking change for all the downstream apps that rely on the library.

However, this particular problem may only manifest in a monorepo setting, I don't know... 

Despite working on this, I'm not a ruby expert by any means, so this may just be me misunderstanding ruby... but it does seem weird that dropping `::` from a library causes problems elsewhere in that same repo... implying there's a logic change by doing that, not just a style change.

**Comment by jeffwidman** [comment]
@koic any feedback given the additional information I provided in ☝️ ?

**Comment by ekohl** [comment]
I also found that this cop is not safe. In https://github.com/theforeman/foreman-installer/pull/883/commits/63c906c1d5715b6a9a2dbbfc798b95ca7ef45cb9#r1343961472 I found an instance.

The reproducer is to have a file:
```ruby
::ENV.delete('x')
```

Then in another you call it with `instance_eval` where there is a `MyModule::ENV`. Without `::` it is interpreted as `MyModule::ENV` instead of `ENV`.

So I'd argue to at least change the cop from safe to unsafe.
